network/src/lib.rs

```diff
@@ -51,32 +51,48 @@ use parking_lot::Mutex;
 /// the justification.
 ///
 /// Note: if no justification is provided the annouce is considered valid.
-pub struct JustifiedBlockAnnounceValidator<B, P> {
+pub struct JustifiedBlockAnnounceValidator<B, P, C> {
 	phantom: PhantomData<B>,
 	polkadot_client: Arc<P>,
+	parachain_client: Arc<C>,
 }
 
-impl<B, P> JustifiedBlockAnnounceValidator<B, P> {
-	pub fn new(polkadot_client: Arc<P>) -> Self {
+impl<B, P, C> JustifiedBlockAnnounceValidator<B, P, C> {
+	pub fn new(polkadot_client: Arc<P>, parachain_client: Arc<C>) -> Self {
 		Self {
 			phantom: Default::default(),
 			polkadot_client,
+			parachain_client,
 		}
 	}
 }
 
-impl<B: BlockT, P> BlockAnnounceValidator<B> for JustifiedBlockAnnounceValidator<B, P>
+impl<B: BlockT, P, C> BlockAnnounceValidator<B> for JustifiedBlockAnnounceValidator<B, P, C>
 where
 	P: ProvideRuntimeApi<PBlock> + HeaderBackend<PBlock>,
 	P::Api: ParachainHost<PBlock>,
+	C: HeaderBackend<B>,
 {
 	fn validate(
 		&mut self,
 		header: &B::Header,
 		mut data: &[u8],
 	) -> Result<Validation, Box<dyn std::error::Error + Send>> {
-		// If no data is provided the announce is valid.
+		// If no data is provided the announce is probably valid
 		if data.is_empty() {
+			// Check if block is one higher than best
+			let best_number = self.parachain_client.info().best_number;
+			let block_number: <<B as BlockT>::Header as HeaderT>::Number = *header.number();
+
+			if !(block_number == best_number + One::one() || block_number == best_number) {
```

Comment:
Okay, I thought again about this. And I made a mistake. 

We should extract the last parachain header that was stored on the relay chain. We get the number of this Parachain header an make sure that:
`block_number == parachain_header_number + 1`

This way we make sure that we support forks on the same height

Reply:
If this condition holds, we need to make sure that there is a justification.